### ISSUE TITLE
canu 1.8 and 1.9 unitigging failure at bogart/AS_BAT_OptimizePosition 

### ISSUE THREAD

**Issue by lorihandley** [marbl/canu#1521]
I have assembled around 20 datasets of PacBio CCS reads using canu 1.8 with great success; however, I am having issues with 3 datasets.  I have tried running canu 1.8 release version and 1.9 (Canu branch v1.9 +316 changes (r9526 d023f631be4cf0a35d6c571daed8cfd46328faa5)) and changing the parameters from ```batOptions=-eg 0.01 -eM 0.01 -dg 6 -db 6 -dr 1 -ca 50 -cp 5``` to ```batOptions=-eg 0.01 -eM 0.01 -dg 4 -db 4 -dr 1 -ca 50 -cp 5```.  Even more strange, one of my problematic datasets has CCS data generated from the CCS3.4 and CCS4.0 algorithms and the CCS4 dataset assembles, but the CCS3.4 does not.  I am running Linux on a sungrid and have tried restarting from the failure point just to make sure it was not a queue issue, but still no success.  The tail of my canu.out for all failing runs looks like this:

```
-- BEGIN ASSEMBLY
--
--
-- Bogart failed, tried 2 times, giving up.
--

ABORT:
ABORT: Canu branch v1.9 +316 changes (r9526 d023f631be4cf0a35d6c571daed8cfd46328faa5)
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT: Disk space available:  94497.144 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (unitigging/4-unitigger/unitigger.err):
ABORT:
ABORT:   optimizePositions()--     Checking convergence.
ABORT:   optimizePositions()--     converged: 8164248 reads
ABORT:   optimizePositions()--     changed:   204486 reads
ABORT:   optimizePositions()--   Recomputing positions, iteration 3, with 16 threads.
ABORT:   optimizePositions()--     Reset zero.
ABORT:   optimizePositions()--     Checking convergence.
ABORT:   optimizePositions()--     converged: 8201085 reads
ABORT:   optimizePositions()--     changed:   167649 reads
ABORT:   optimizePositions()--   Recomputing positions, iteration 4, with 16 threads.
ABORT:   optimizePositions()--     Reset zero.
ABORT:   optimizePositions()--     Checking convergence.
ABORT:   optimizePositions()--     converged: 8212275 reads
ABORT:   optimizePositions()--     changed:   156459 reads
ABORT:   optimizePositions()--   Recomputing positions, iteration 5, with 16 threads.
ABORT:   optimizePositions()--     Reset zero.
ABORT:   optimizePositions()--     Checking convergence.
ABORT:   optimizePositions()--     converged: 8233794 reads
ABORT:   optimizePositions()--     changed:   134940 reads
ABORT:   optimizePositions()--   Expanding short reads with 16 threads.
ABORT:   optimizePositions()--   Updating positions.
ABORT:   optimizePositions()--   Finished.
ABORT:   splitDiscontinuous()-- Tested 25954 tigs, split 1 tig into 2 new tigs.
ABORT:
ABORT:   ==> PLACE CONTAINED READS.
ABORT:
ABORT:   computeErrorProfiles()-- Computing error profiles for 577029 tigs, with 16 threads.
ABORT:   computeErrorProfiles()-- Finished.
ABORT:
ABORT:   placeContains()-- placing 4090718 contained and 1184717 unplaced reads, with 16 threads.
ABORT:   placeContains()-- Placed 4053366 contained reads and 501198 unplaced reads.
ABORT:   placeContains()-- Failed to place 37352 contained reads (too high error suspected) and 683519 unplaced reads (lack of overlaps suspected).
ABORT:   optimizePositions()-- Optimizing read positions for 8368734 reads in 577029 tigs, with 16 threads.
ABORT:   optimizePositions()--   Allocating scratch space for 8368734 reads (523045 KB).
ABORT:   optimizePositions()--   Initializing positions with 16 threads.
ABORT:   bogart: bogart/AS_BAT_OptimizePositions.C:260: void Unitig::optimize_initPlace(uint32, optPos*, optPos*, bool, std::set<unsigned int>&, bool): Assertion `cnt > 0' failed.
ABORT:
ABORT:   Failed with 'Aborted'; backtrace (libbacktrace):
ABORT:   utility/system-stackTrace.C::89 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   bogart/AS_BAT_OptimizePositions.C::260 in _ZN6Unitig18optimize_initPlaceEjP6optPosS1_bRSt3setIjSt4lessIjESaIjEEb()
ABORT:   bogart/AS_BAT_OptimizePositions.C::519 in _ZN9TigVector17optimizePositionsEPKcS1_._omp_fn.0()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   Aborted (core dumped)
```

**Comment by skoren** [comment]
I think this got fixed in 1.9 slightly after your code checkout. Try updating the 1.9 branch you have to the latest, recompile, and re-run.

**Comment by skoren** [comment]
And if you pulled 1.9 between Oct 16-now, there was a bug with cnt > 0 introduced during that time so make sure you pull and re-build now.

**Comment by lorihandley** [comment]
Yes, I had pulled the latest as of this morning and restarted one of my
assemblies, but will pull/rebuild/restart again now.  Thank you for letting
me know!

**Comment by skoren** [comment]
Closing as the latest code resolves it per the user (see comment in #1520).
